torchvision/prototype/models/_api.py

```diff
@@ -78,41 +84,37 @@ def __getattr__(self, name):
         return super().__getattr__(name)
 
 
-def get_weight(fn: Callable, weight_name: str) -> WeightsEnum:
+def get_weight(name: str) -> WeightsEnum:
```

Comment:
I'm wondering: shoudn't this return a `Weight` instance, instead of `WeightsEnum`?
Same for `from_str` (and in the return section below).

Reply:
I think we need to return a `WeightsEnum` value. That is the value of an Enum which maintains information about the class it comes from (`ResNet50_Weights`). Returning a `Weights` loses the information necessary to validate that the right type of weights were passed to the method.